### ISSUE TITLE
MenuBar Global Menu queue_free() on PopupMenu causes crash during runtime

### ISSUE THREAD

**Issue by mr-dreich** [godotengine/godot#89066]
### Tested versions

Happens only in 4.3.dev4, works fine in 4.3.dev3 and 4.2 stable.

### System information

Godot v4.3.dev4 - macOS 14.3.0 - GLES3 (Compatibility) - Apple M1 - Apple M1 (8 Threads)

### Issue description

This error does not happen when I run it in Windows.

I can open the project fine, and even run it. But after a while it will crash at runtime in each Renderer. Forward+ and Mobile produce an actual error in the Editor and Compatibility just crashes and I have to refer to the godot.log. The title is the error that Forward and Mobile produce, and here is the log from Compatibility:

```
Godot Engine v4.3.dev4.official.df78c0636 - https://godotengine.org
OpenGL API 4.1 Metal - 88 - Compatibility - Using Device: Apple - Apple M1

USER ERROR: FATAL: Index p_index = 2 is out of bounds (size() = 2).
   at: get (./core/templates/cowdata.h:205)

================================================================
handle_crash: Program crashed with signal 5
Engine version: Godot Engine v4.3.dev4.official (df78c0636d79c9545a283e0e2a926d623998cc27)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x0000000187c11a24 _sigtramp + 56
[2] msdfgen::OverlappingContourCombiner<msdfgen::MultiAndTrueDistanceSelector>::distance() const
[3] ItemList::get_v_scroll_bar()
[4] Node::get_tree() const
[5] MultiplayerAPI::is_server()
[6] PlaceholderExtensionInstance::placeholder_class_recreate_instance(void*, void*)
[7] Node::atr_n(String, StringName const&, int, StringName) const
[8] Node::atr_n(String, StringName const&, int, StringName) const
[9] RendererCompositorRD::_create_current()
[10] void rx::mtl::GetMatrixUniformMetal<unsigned int>(unsigned int, unsigned int*, unsigned int const*, bool)
[11] RendererCompositorRD::_create_current()
[12] 12  dyld                                0x00000001878610e0 start + 2360
-- END OF BACKTRACE --
================================================================
```

4.3.dev3 works fine. 

I thought this might have something to do with a shader so I removed it and it still crashes.

I only found this issue because I wanted to try some of the new global menu additions available in 4.3 that 4.2 does not have so this is a fork of my main from 4.2 which works 100% perfectly.

### Steps to reproduce

Open the Editor, press Run, wait until crash.

### Minimal reproduction project (MRP)

I will provide if necessary, but this is a long time project that I would prefer not be made public.

**Comment by clayjohn** [comment]
Unfortunately you are going to have to do a little bit of investigation before we can help you. 

Ideally, you would compile the engine with debug symbols so that you can get a proper stack trace. The one you have posted just contains random information and doesn't help us identify the issue. 

If you can't do that, you will have to trim down your project to be as small as possible while still recreating the error. We call that an MRP.

Without one of those we can't really do anything as we don't have enough information to go off of. 

The title of your post seems unrelated to the content of the post. But the error message that is in the title usually occurs when you try to render an object using a ViewportTexture that points to the Viewport you are trying to render the object into. Doing so is not allowed, hence the error. If you are getting that error consistently, I would start by trying to find places where you might be doing that and fixing them.

**Comment by mr-dreich** [comment]
@clayjohn I'll compile with symbols once I get home tonight and report back!

I do have a separate viewport that I eventually take the texture from, but I can't even get to the scene that happens in. If this isn't allowed, its strange it works fine in Windows and in dev3 works both MacOS and windows.

**Comment by clayjohn** [comment]
> I do have a separate viewport that I eventually take the texture from, but I can't even get to the scene that happens in. If this isn't allowed, its strange it works fine in Windows and in dev3 works both MacOS and windows.

The error could be a red herring.

Let me know how your investigation goes!

**Comment by mr-dreich** [comment]
@clayjohn Thanks for your advice through this! [Here is the MacOS ](https://pastebin.com/raw/Nvs5Whmd)crash log generated after compiling from source with debug symbols enabled.

Luckily, this runs fine on dev3 and in its actual development in 4.2, If you need me to do any more testing please let me know. Again, I would be willing to let a trusted user look at my project if need be, but I have spent a lot of time and money on this and its currently for sale on itch.

**Comment by clayjohn** [comment]
Maybe https://github.com/godotengine/godot/pull/86322 is related. Its the only PR touching MenuBar and it seems targeted at macOS. Of course, the issue could be totally unrelated 

CC @bruvzg

**Comment by mr-dreich** [comment]
I'll disable or remove my MenuBar setup and report back tomorrow.

And I think this is an issue with MacOS, as it works fine on Windows and Linux (Tested it when I got home). So something between dev3 and dev4 for MacOS may be at fault.

**Comment by mr-dreich** [comment]
I have removed the MenuBar and it works just fine. Seems the issue is related to what @clayjohn linked.

**Comment by mr-dreich** [comment]
I have an update, after finding out that it was a problem with the MenuBar, I went through and started commenting out my MenuBar code and I found out what is causing the issue.

```
func _ready() -> void:
	if MBar.is_native_menu():
		%MenuBackground.hide()
		%System.queue_free()  <--- This causes the crash
```

The part in the example is removing the popup menu that would be there if you were in Windows or another non-global menu OS, all the items get moved to the "_apple" global menu and then this is removed since its no longer needed.

**Comment by bruvzg** [comment]
> The part in the example is removing the popup menu that would be there if you were in Windows or another non-global menu OS, all the items get moved to the "_apple" global menu and then this is removed since its no longer needed.

 I can't reproduce any issues with removing native menus from `MenuBar`, please provide minimal project. It's probably caused by some specific condition, and it's not entirely clear what exactly your project is doing.
